apps/admin-gui/src/app/shared/components/dialogs/edit-application-form-item-dialog/edit-application-form-item-dialog.component.ts

```diff
@@ -145,6 +150,28 @@ export class EditApplicationFormItemDialogComponent implements OnInit {
     this.getOptions();
   }
 
+  loadWarning(itemType: ItemType): void {
+    this.warningMessage = '';
+    const hiddenDependencyForItem = this.data.allItems.find(
+      (item) => item.hiddenDependencyItemId === this.data.applicationFormItem.id
+    );
+    const disabledDependencyForItem = this.data.allItems.find(
+      (item) => item.disabledDependencyItemId === this.data.applicationFormItem.id
+    );
+    if (hiddenDependencyForItem || disabledDependencyForItem) {
+      this.displayWarningForType = itemType;
+      this.warningMessage = this.translate.instant(
+        'DIALOGS.APPLICATION_FORM_EDIT_ITEM.DEPENDENCY_WARNING_MESSAGE',
+        {
+          dependency: hiddenDependencyForItem ? 'hidden' : 'disabled',
+          shortname: hiddenDependencyForItem
+            ? hiddenDependencyForItem.shortname
+            : disabledDependencyForItem.shortname,
+        }
+      );
+    }
```

Comment:
Little nit-picky but this way there are two unnecessary if-else conditions. Same goes for the deletion part.